hazelcast/src/main/java/com/hazelcast/client/config/ClientConfig.java

```diff
@@ -919,6 +921,29 @@ public ClientConfig setUserContext(ConcurrentMap<String, Object> userContext) {
         return userContext;
     }
 
+    /**
+     * This feature reduces number of hops and increase performance for smart clients.
+     * It is enabled by default for smart clients. For unisocket clients this feature can not be enabled.
```

Comment:
we can define what happens if set for dummy client, better to clarify `feature can not be enabled` wording.